tests/unit/s2n_key_update_test.c

```diff
@@ -248,37 +265,46 @@ int main(int argc, char **argv)
         /* Skip test if libcrypto doesn't support the cipher */
         if (s2n_chacha20_poly1305.is_available()) {
             struct s2n_connection *conn;
-            uint8_t data_size = 1;
             EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
             conn->actual_protocol_version = S2N_TLS13;
             /* Setting cipher suite to suite that does not have an encryption limit */
             conn->secure.cipher_suite = &s2n_tls13_chacha20_poly1305_sha256;
-            
-            EXPECT_EQUAL(conn->key_update_pending, 0);
-            conn->encrypted_bytes_out = UINT64_MAX;
+            struct s2n_blob sequence_number = {0};
+            EXPECT_SUCCESS(s2n_blob_init(&sequence_number, conn->secure.server_sequence_number, S2N_TLS_SEQUENCE_NUM_LEN));
 
-            EXPECT_SUCCESS(s2n_check_key_limits(conn, data_size));
             EXPECT_EQUAL(conn->key_update_pending, 0);
+            for (int i = 0; i < S2N_TLS_SEQUENCE_NUM_LEN; i++) {
+                conn->secure.server_sequence_number[i] = UINT8_MAX;
+            }
+
+            EXPECT_SUCCESS(s2n_check_key_limits(conn, &sequence_number));
+
+            EXPECT_EQUAL(conn->key_update_pending, false);
 
             EXPECT_SUCCESS(s2n_connection_free(conn)); 
         }
 
         /* Key update NOT triggered when cipher suite does not have encryption limit and
-         * when encrypted_bytes_out exactly equals UINT64_MAX
+         * when record limit exactly equals UINT64_MAX
```

Comment:
if (in theory) sequence number reaches uint64_max, shouldn't s2n kill the connection or send another keyupdate to reset sequence numbers?

Reply:
s2n does kill the connection in this case, that happens in the record_write() function. This test just checks to make sure that a key update is not triggered.